trace/impl/include/trace_impl.h

```diff
@@ -0,0 +1,46 @@
+#include "trace.h"
```

Comment:
Note to reviewers: Much of this file was copied untouched from the original `trace.c`.